mailbox/api/src/test/java/org/apache/james/mailbox/events/ErrorHandlingContract.java

```diff
@@ -82,6 +84,28 @@ default void retryingIsNotAppliedForKeyRegistrations() {
             .isEmpty();
     }
 
+    @Test
+    default void deadLettersIsNotAppliedForKeyRegistrations() throws Exception {
+        EventCollector eventCollector = eventCollector();
+
+        doThrow(new RuntimeException())
+            .doThrow(new RuntimeException())
+            .doThrow(new RuntimeException())
+            .doThrow(new RuntimeException())
+            .doCallRealMethod()
+            .when(eventCollector).event(EVENT);
+
+        eventBus().register(eventCollector, KEY_1);
+        eventBus().dispatch(EVENT, ImmutableSet.of(KEY_1)).block();
+
+        TimeUnit.SECONDS.sleep(1);
+        SoftAssertions.assertSoftly(softly -> {
+            softly.assertThat(eventCollector.getEvents()).isEmpty();
+            softly.assertThat(deadLetter().failedEventIds(GROUP_A).toIterable())
```

Comment:
You should assert `no group is stored in dead-letter` because we know for sure nothing happened in groupA

Reply:
very right!